### PR TITLE
Add labels to spec

### PR DESCRIPTION
**Pull request by wilzbach**
Labels are nice and necessary if you want to direct a reader to a specific spot at the D spec.

This contains the remaining bits of my pass through the D spec:

```
ls | parallel -j1 "echo {}; cat {} | grep -E 'H(2|3|4|5)'" | grep -v LNAME2
```

See also:
- https://github.com/dlang/dlang.org/pull/1677
- https://github.com/dlang/dlang.org/pull/1675
- https://github.com/dlang/dlang.org/pull/1674

### CODE REVIEW

File: spec/property.dd [hunk 1]
```diff
@@ -65,7 +65,7 @@ $(THEAD Property, Description)
 $(TROW $(RELATIVE_LINK2 classinfo, $(D .classinfo)), Information about the dynamic type of the class)
 )
 
-$(SECTION3 $(LNAME2 init, .init) Property,
+$(SECTION3 $(LNAME2 init, .init Property),
```
Comment (wilzbach) [inline spec/property.dd]: This unifies this page (same headings contain "Property" in the link, some didn't).

File: spec/ddoc.dd [hunk 2]
```diff
@@ -1043,7 +1043,7 @@ $(P
     cross-site scripting).
 )
 
-$(H2 Links to D documentation generators)
+$(H2 $(LNAME2 links_to_d_documentation_generators, Links to D documentation generators))
 
 $(P
     A list of current D documentation generators which use Ddoc
```
Comment (ZombineDev) [inline spec/ddoc.dd]: Can you give "Comparing D Immutable and Const with C++ Const" the same treatment?

File: spec/cpp_interface.dd [hunk 3]
```diff
@@ -315,7 +315,7 @@ instead disabled on the D side; an alternative would be to reimplement the
 constructor in D. See the $(DDSUBLINK spec/cpp_interface, lifetime-management,
 section below on lifetime management) for more information.)
 
-$(H3 Using D Classes From C++)
+$(H3 $(LNAME2 using_d_classes_from_cpp, Using D Classes From C++))
 
     $(P Given D code like:)
```
Comment (ZombineDev) [inline spec/cpp_interface.dd]: Can you give "Comparing D Immutable and Const with C++ Const" the same treatment?

Comment (ZombineDev) [inline spec/cpp_interface.dd]: Can you give "Comparing D Immutable and Const with C++ Const" the same treatment?

**Comment by wilzbach** [comment]
> modulo one missed label.

Nice catch. I didn't think about checking for `SECTION2`.

**Pull request merged**